services/sources/dspace_service.py

```diff
@@ -0,0 +1,143 @@
+from datetime import datetime, timedelta, timezone
+import os
+import requests
+from io import BytesIO
+from lxml import etree
+from constants.get_constants import get_constants
+from logger import create_log
+from mappings.base_mapping import MappingError
+from managers import S3Manager, RabbitMQManager
+from mappings.xml import XMLMapping
+from .source_service import SourceService
+
+logger = create_log(__name__)
+
+
+class DSpaceService(SourceService):
+    OAI_NAMESPACES = {
+        'oai_dc': 'http://www.openarchives.org/OAI/2.0/oai_dc/',
+        'dc': 'http://purl.org/dc/elements/1.1/',
+        'datacite': 'https://schema.datacite.org/meta/kernel-4.1/metadata.xsd',
+        'oapen': 'http://purl.org/dc/elements/1.1/',
+        'oaire': 'https://raw.githubusercontent.com/rcic/openaire4/master/schemas/4.0/oaire.xsd'
+    }
+
+    def __init__(self, base_url, SourceMapping: XMLMapping):
```

Comment:
This is fun cuz I've actually never done this in python so we are learning ha!

I think when we say that the SourceMapping type is XMLMapping, we are expecting an actual object that is in instance of XMLMapping. 

Checkout typing for Python: https://docs.python.org/3/library/typing.html

I think we could do type[XMLMapping] or the like here.